### PR TITLE
FIX Respect SS_BASE_URL scheme in CLI environment

### PR DESCRIPTION
**Pull request by s-kerdel**
Additionally set _SERVER variables for HTTPS and SSL to respect SS_BASE_URL scheme when executing builds and tasks through CLI. This should solve base tags not being provided with the correct HTTP scheme. This is important to resolve mixed content issues and insecure requests.

## Parent issue
- https://github.com/silverstripe/silverstripe-framework/issues/10615

### CODE REVIEW

**Review by GuySartorelli — APPROVED** [review]
This seems okay to me - I notice some discussion about the host in the issue, but it seems like the host is already being set correctly and any change to the way that's handled could be an additional enhancement if we discover that it would be useful.
This PR does seem to resolve a genuine issue in and off itself.

I'd like another pair of eyes over this before merging @silverstripe/core-team 

Also, this should probably target the `4.12` branch so that it can be released in a patch release rather than waiting for the `4.13` minor release.

**Review by dhensby — APPROVED** [review]

**Comment by GuySartorelli** [comment]
I've changed the base of this to `4.12` so it can be in a patch release. I'll merge once CI is green again.

**Pull request merged**
